L1Trigger/L1THGCal/plugins/veryfrontend/HGCalVFEProcessorSums.cc

```diff
@@ -74,12 +76,15 @@ HGCalVFEProcessorSums::run(const HGCEEDigiCollection& ee,
 
   vfeLinearizationImpl_.linearize(dataframes, linearized_dataframes);
   vfeSummationImpl_.triggerCellSums(*geometry_, linearized_dataframes, payload);  
+  vfeCompressionImpl_.compress(payload, compressed_payload);
   
   // Transform map to trigger cell vector vector<HGCalTriggerCell>
   for(const auto& id_value : payload)
   { 
     if (id_value.second>0){
       l1t::HGCalTriggerCell triggerCell(reco::LeafCandidate::LorentzVector(), id_value.second, 0, 0, 0, id_value.first.rawId());
+      triggerCell.setCompressedCharge(compressed_payload[id_value.first][0]);
+      triggerCell.setUncompressedCharge(compressed_payload[id_value.first][1]);
```

Comment:
It would actually need to be the opposite here:
```cpp
l1t::HGCalTriggerCell triggerCell(reco::LeafCandidate::LorentzVector(),compressed_payload[id_value.first][1], 0, 0, 0, id_value.first.rawId());
triggerCell.setUncompressedCharge(id_value.second);
```
Because the `hwPt` is actually what is used later on to compute the final trigger cell energy, so it should be the value after compression on the linear scale.
While the uncompressed value should just be kept to keep track of the original charge, in the `uncompressedCharge` variable.

Reply:
Ahhh I see. Now it makes much more sense now...

I am also thinking of renaming compressedCharge variable to something like chargeCode etc to make it clearer that we are referring to the compressed code space instead of the charge value space.